pkg/git/gitclient/gitclient.go

```diff
@@ -268,11 +270,23 @@ func (g *GitClient) Branch(name string) error {
 
 func (g *GitClient) ValidateRemoteExists(ctx context.Context) error {
 	logger.V(3).Info("Validating git setup", "repoUrl", g.RepoUrl)
-	remote := g.Client.NewRemote(g.RepoUrl, gogit.DefaultRemoteName)
-	// Check if we are able to make a connection to the remote by attempting to list refs
-	_, err := g.Client.ListWithContext(ctx, remote, g.Auth)
-	if err != nil {
-		return fmt.Errorf("connecting with remote %v for repository: %v", gogit.DefaultRemoteName, err)
+
+	repositoryURL, _ := url.Parse(g.RepoUrl)
```

Comment:
should we be handling this error?

Reply:
It's already handled in gitfactory before this step.